src/components/buttons/link-button/link-button.spec.js

```diff
@@ -1,141 +1,43 @@
-        <Text.Body>{'Accessibility text'}</Text.Body>
-      )
-    ).toBe(true);
-  });
-
-  it('should render the icon', () => {
-    expect(wrapper).toRender(PlusThinIcon);
-  });
-
-  describe('with data-* props', () => {
-    beforeEach(() => {
-      props = createTestProps({
-        'data-track-component': 'LinkButton',
-        'data-track-label': 'LinkButton',
-        'data-track-event': 'click',
-        'data-test': 'link-button',
-      });
-      wrapper = shallow(<LinkButton {...props} />);
-    });
-    it('should apply given `data-track-component` to LinkButton', () => {
-      expect(wrapper.find('Link')).toHaveProp(
-        'data-track-component',
-        expect.stringMatching('LinkButton')
-      );
-    });
-
-    it('should apply given `data-track-event` to AccessibleButton', () => {
-      expect(wrapper.find('Link')).toHaveProp(
-        'data-track-event',
-        expect.stringMatching('click')
-      );
-    });
-    it('should apply given `data-track-label` to AccessibleButton', () => {
-      expect(wrapper.find('Link')).toHaveProp(
-        'data-track-label',
-        expect.stringMatching('LinkButton')
-      );
-    });
-    it('should apply given `data-test` to AccessibleButton', () => {
-      expect(wrapper.find('Link')).toHaveProp(
-        'data-test',
-        expect.stringMatching('link-button')
-      );
-    });
-  });
-
-  describe('without data-* props', () => {
-    beforeEach(() => {
-      props = createTestProps();
-      wrapper = shallow(<LinkButton {...props} />);
-    });
-    it('should apply default `data-track-component` to LinkButton', () => {
-      expect(wrapper.find('Link')).toHaveProp(
-        'data-track-component',
-        expect.stringMatching('LinkButton')
-      );
+  it('should prevent the navigation when "disabled"', async () => {
+    const { getByLabelText, history } = render(
+      <LinkButton {...props} isDisabled={true} />
+    );
+    fireEvent.click(getByLabelText('test-button'));
+    await wait(() => {
+      expect(history.location.pathname).toBe('/');
     });
   });
-
-  describe('if disabled', () => {
-    beforeEach(() => {
-      props = createTestProps({ isDisabled: true });
-      wrapper = shallow(<LinkButton {...props} />);
-    });
-    it('renders a <Link> with disabled class', () => {
-      expect(wrapper.find('Link')).toHaveClassName('disabled');
-    });
-
-    describe('onClick', () => {
-      const testEvent = createEvent();
-      beforeEach(() => {
-        wrapper.find('Link').simulate('click', testEvent);
-      });
-      it('renders with a onClick prop', () => {
-        expect(wrapper.find('Link')).toHaveProp('onClick');
-      });
-      it('when `onClick` is fired, event default is prevented', () => {
-        expect(testEvent.preventDefault).toHaveBeenCalled();
-      });
-    });
-
-    it('renders the icon with "grey" theme', () => {
-      expect(wrapper.find(PlusThinIcon)).toHaveProp('theme', 'grey');
-    });
```

Comment:
Nice that we move this to rtl. Could this have been another PR though or was it required in the refactor to emotion?

Reply:
Yes and no. But since I changed some stuff internal to the components, I would have had to fix all the tests. This was way quicker...

Reply:
Makes total sense.